### PR TITLE
Add links to std::char::REPLACEMENT_CHARACTER from docs.

### PR DESCRIPTION
**Pull request by frewsxcv**
There are a few places where we mention the replacement character in the
docs, and it could be helpful for users to utilize the constant which is
available in the standard library, so let’s link to it!

### CODE REVIEW

**Comment by frewsxcv** [comment]
https://doc.rust-lang.org/nightly/std/char/constant.REPLACEMENT_CHARACTER.html

File: src/libstd/ffi/c_str.rs [hunk 1]
```diff
@@ -1175,9 +1175,9 @@ impl CStr {
     /// If the contents of the `CStr` are valid UTF-8 data, this
     /// function will return a [`Cow`]`::`[`Borrowed`]`(`[`&str`]`)`
     /// with the the corresponding [`&str`] slice. Otherwise, it will
-    /// replace any invalid UTF-8 sequences with `U+FFFD REPLACEMENT
-    /// CHARACTER` and return a [`Cow`]`::`[`Owned`]`(`[`String`]`)`
-    /// with the result.
+    /// replace any invalid UTF-8 sequences with
+    /// [`U+FFFD REPLACEMENT CHARACTER`][U+FFFD] and return a
```
Comment (sfackler) [inline src/libstd/ffi/c_str.rs]: Looks like we're a bit inconsistent between "`U+FFFD REPLACEMENT CHARACTER`" and "U+FFFD REPLACEMENT CHARACTER".

Comment (frewsxcv) [reply src/libstd/ffi/c_str.rs]: What's your opinion? I don't feel strongly

Comment (sfackler) [reply src/libstd/ffi/c_str.rs]: I don't really feel strongly either.

Comment (frewsxcv) [reply src/libstd/ffi/c_str.rs]: Made consistent in the latest force push

File: src/liballoc/string.rs [hunk 2]
```diff
@@ -621,14 +622,15 @@ impl String {
     }
 
     /// Decode a UTF-16 encoded slice `v` into a `String`, replacing
-    /// invalid data with the replacement character (U+FFFD).
+    /// invalid data with [the replacement character (`U+FFFD`)][U+FFFD].
```
Comment (fbstj) [inline src/liballoc/string.rs]: was this missed in the push to `U+FFFD REPLACEMENT CHARACTER` everywhere (the above comment)?

Comment (frewsxcv) [reply src/liballoc/string.rs]: Not missed, I intentionally didn't change this instance. Not sure I see the value in making all of these cases exactly identical

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**